packages/contracts-bedrock/test/L2/LiquidityController.t.sol

```diff
@@ -0,0 +1,272 @@
+
+    /// @notice Emitted when liquidity is burned
+    event LiquidityBurned(address indexed minter, uint256 amount);
+
+    /// @notice Test setup.
+    function setUp() public virtual override {
+        enableCustomGasToken();
+        super.setUp();
+    }
+
+    /// @notice Tests that contract is set up correctly.
+    function test_setup_succeeds() public view {
+        assertEq(liquidityController.version(), "1.0.0");
+        assertEq(liquidityController.gasPayingTokenName(), "Custom Gas Token");
+        assertEq(liquidityController.gasPayingTokenSymbol(), "CGT");
+    }
+
+    /// @notice Shared modifier to authorize a minter.
+    modifier isAuthorizedMinter(address _minter) {
+        // Authorize the minter
+        stdstore.target(address(liquidityController)).sig(liquidityController.minters.selector).with_key(_minter)
+            .checked_write(true);
+        _;
+    }
+}
+
+/// @title LiquidityController_AuthorizeMinter_Test
+/// @notice Tests the `authorizeMinter` function of the `LiquidityController` contract.
+contract LiquidityController_AuthorizeMinter_Test is LiquidityController_TestInit {
+    /// @notice Tests that the authorizeMinter function can be called by the owner.
+    function testFuzz_authorizeMinter_fromOwner_succeeds(address _minter) public {
+        // Expect emit MinterAuthorized event
+        vm.expectEmit(address(liquidityController));
+        emit MinterAuthorized(_minter);
+        // Call the authorizeMinter function with owner as the caller
+        vm.prank(IProxyAdmin(Predeploys.PROXY_ADMIN).owner());
+        liquidityController.authorizeMinter(_minter);
+
+        // Assert minter is authorized
+        assertTrue(liquidityController.minters(_minter));
+    }
+
+    /// @notice Tests that the authorizeMinter function reverts when called by non-owner.
+    function testFuzz_authorizeMinter_fromNonOwner_fails(address _caller, address _minter) public {
+        vm.assume(_caller != IProxyAdmin(Predeploys.PROXY_ADMIN).owner());
+
+        // Call the authorizeMinter function with non-owner as the caller
+        vm.prank(_caller);
+        vm.expectRevert(Unauthorized.selector);
+        liquidityController.authorizeMinter(_minter);
+
+        // Assert minter is not authorized
+        assertFalse(liquidityController.minters(_minter));
+    }
+}
+
+/// @title LiquidityController_DeauthorizeMinter_Test
+/// @notice Tests the `deauthorizeMinter` function of the `LiquidityController` contract.
+contract LiquidityController_DeauthorizeMinter_Test is LiquidityController_TestInit {
+    using stdStorage for StdStorage;
+
+    /// @notice Tests that the deauthorizeMinter function can be called by the owner.
+    function testFuzz_deauthorizeMinter_fromOwner_succeeds(address _minter) public {
+        // Set minter to authorized
+        stdstore.target(address(liquidityController)).sig(liquidityController.minters.selector).with_key(_minter)
```

Comment:
Nit - there are a few places where the helper function could be used but the checked write is used instead